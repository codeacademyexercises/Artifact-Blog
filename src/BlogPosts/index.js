export default [{ title: 'The future of abstract art and the culture ...', description: 'Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...' }, { title: 'How painting outdoors can transform your ', description: 'Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...' }, { title: '10 young painters you need to know', description: 'Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...' },{title: 'Tips for creating the perfect art', description: 'Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...'},{title: 'Tips for creating the perfect art', description: 'Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...'},{title: 'Tips for creating the perfect art', description: 'Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...'},]
