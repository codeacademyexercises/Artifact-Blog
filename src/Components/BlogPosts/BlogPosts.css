.blogPosts {
    /* height: 100%;
    width: 100%; */
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
}

.PostContainer {
    /* height: 70%; */
    width: 80%;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: flex-start;
    flex-wrap: wrap;
}

