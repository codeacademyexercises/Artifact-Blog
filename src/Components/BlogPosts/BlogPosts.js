import React, { Component } from 'react';
import './BlogPosts.css';
import {BrowserRouter as Router, Route,Link} from 'react-router-dom';
import BlogCard from '../BlogCard/BlogCard.js';
import Submit from '../Submit/Submit.js';
const fs = require('fs');

class BlogPosts extends Component {
    blogPostsJSON = require('../../BlogPosts');
    render(){
            console.log(this.blogPostsJSON);
        let display = [];
        this.blogPostsJSON.default.forEach((postJSON)=>{
                display.push(<BlogCard title={postJSON.title} description={postJSON.description}/>);
        });
        return (
                <div className="blogPosts">
                        <div className="PostContainer">
                        {display}
                        </div>
                        <Submit className="submit"/>
                </div>
        );
    };
}
export default BlogPosts;