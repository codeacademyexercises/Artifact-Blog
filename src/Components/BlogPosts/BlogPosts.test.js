import React from 'react';
import { shallow } from 'enzyme';
import BlogPosts from './BlogPosts.js';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router, Route} from 'react-router-dom';

describe('<BlogPosts /> ',()=>{
    it('snapshot test',()=>{
        const tree = renderer.create(<Router><BlogPosts /></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});