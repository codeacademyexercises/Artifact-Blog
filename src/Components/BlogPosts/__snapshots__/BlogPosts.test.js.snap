// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPosts />  snapshot test 1`] = `
<div>
  <div
    className="posts"
  >
    <div>
      <img
        height="210px"
        src="abstract.png"
        width="240px"
      />
      <h3>
        The future of abstract art and the culture ...
      </h3>
      <p>
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </p>
      <span>
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "clapping.svg",
            }
          }
          width="20px"
        />
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-red.svg",
            }
          }
          width="20px"
        />
      </span>
    </div>
    <div>
      <img
        height="210"
        src="outdoor.png"
        width="240"
      />
      <h3>
        How painting outdoors can transform your 
      </h3>
      <p>
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </p>
      <span>
        <img
          alt=""
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "clapping.svg",
            }
          }
          width="20px"
        />
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-red.svg",
            }
          }
          width="20px"
        />
      </span>
    </div>
    <div>
      <img
        height="210"
        src="young-painters.png"
        width="240"
      />
      <h3>
        10 young painters you need to know
      </h3>
      <p>
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </p>
      <span>
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "clapping.svg",
            }
          }
          width="20px"
        />
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-red.svg",
            }
          }
          width="20px"
        />
      </span>
    </div>
  </div>
  <div
    className="posts"
  >
    <div>
      <img
        height="210"
        src="perfect-art.png"
        width="240"
      />
      <h3>
        Tips for creating the perfect art
      </h3>
      <p>
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </p>
      <span>
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "clapping.svg",
            }
          }
          width="20px"
        />
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-red.svg",
            }
          }
          width="20px"
        />
      </span>
    </div>
    <div>
      <img
        height="210"
        src="inspired.png"
        width="240"
      />
      <h3>
        Stay inspired: top artists blog ...
      </h3>
      <p>
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </p>
      <span>
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "clapping.svg",
            }
          }
          width="20px"
        />
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-red.svg",
            }
          }
          width="20px"
        />
      </span>
    </div>
    <div>
      <img
        height="210"
        src="reason.png"
        width="240"
      />
      <h3>
        5 reasons why everyone should try an...
      </h3>
      <p>
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </p>
      <span>
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "clapping.svg",
            }
          }
          width="20px"
        />
        <img
          height="20px"
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "heart-red.svg",
            }
          }
          width="20px"
        />
      </span>
    </div>
  </div>
  <a
    href="/form"
    onClick={[Function]}
  >
    <input
      name="post"
      type="button"
      value="Post"
    />
  </a>
</div>
`;
